### PR TITLE
Make more space for the navigation links at smaller screen resolutions

### LINKED ISSUE #10254: Navigation links wrap at different screen resolutions
This is something I noticed when working on https://github.com/mozilla/addons-frontend/issues/10252, and is a regression from https://github.com/mozilla/addons-frontend/issues/9078 (I believe).

When we bumped up our fonts we ended up with a situation where our navigation links wrap at particular screen resolutions. This is because our current layout is based on certain breakpoints, and those are no longer valid with these new font sizes.

Here are a couple of examples from near the breakpoints:

900px wide:

Screen Shot 2021-03-22 at 4 20 47 PM

500px wide:

Screen Shot 2021-03-22 at 4 21 38 PM

I can think of a couple of ways of addressing this:

1. Change our breakpoints to be at larger resolutions. I think this makes the most sense as these are probably not the only things that look less than ideal with the new font sizes and the old breakpoints.
2. Reduce the size of the font for these particular menu items.

As mentioned, I am leaning towards number 1, which would involve bumping our `extraLarge` breakpoint from `900px` to `1050px` and our `medium` breakpoint from `500px` to `600px`. TBH, I'm not sure where our original breakpoints came from, and if this is a reasonable thing to do. It also might make sense to bump the other ones as well, but that is not needed to address this bug.

WDYT @muffinresearch @verdi et al?

### PR DESCRIPTION
**Pull request by bobsilverberg**
Fixes #10254 

This fixes the wrapping of the navigation links at `500px` and `900px`, as well as an issue with the "More..." menu.

<details>
<summary>At `500px`</summary>

Before:
Screen Shot 2021-03-23 at 9 57 47 AM

After:
Screen Shot 2021-03-23 at 9 57 09 AM
</details>

<details>
<summary>At `900px`</summary>

Before:
Screen Shot 2021-03-23 at 9 58 25 AM

After:
Screen Shot 2021-03-23 at 9 58 50 AM
</details>

### CODE REVIEW

File: src/amo/components/SectionLinks/styles.scss [hunk 1]
```diff
@@ -16,11 +16,16 @@
   color: $grey-40;
   display: inline-block;
   font-weight: 400;
-  margin: 0 12px;
+  margin: 0 8px;
   outline: none;
+  overflow-wrap: break-word;
```
Comment (willdurand) [inline src/amo/components/SectionLinks/styles.scss:21]: I suggest we add a comment to explain why `break-word` is needed here, given that we should only rely on the global/default `overflow-wrap: anywhere` rule in this project.

**Review by diox — APPROVED** [review]
It's better, but looks kinda weird to me now because the width of the search field is reduced... even when the links aren't there, so it looks a bit jarring (try with `/android/` where the links aren't shown)

**Comment by diox** [comment]
Also actually this makes it take more vertical space...

**Comment by bobsilverberg** [comment]
> It's better, but looks kinda weird to me now because the width of the search field is reduced... even when the links aren't there, so it looks a bit jarring (try with `/android/` where the links aren't shown)

I think I can address this.

**Comment by bobsilverberg** [comment]
@diox I addressed the issue with the search field on mobile:

Mobile @ `500px`:
Screen Shot 2021-03-23 at 1 57 46 PM

Desktop @ `500px`:
Screen Shot 2021-03-23 at 1 59 14 PM

**Pull request merged**